core/src/main/java/org/apache/accumulo/core/client/impl/ScannerOptions.java

```diff
@@ -244,4 +247,20 @@ public void setBatchTimeout(long timeout, TimeUnit timeUnit) {
   public long getBatchTimeout(TimeUnit timeUnit) {
     return timeUnit.convert(batchTimeOut, TimeUnit.MILLISECONDS);
   }
+
+  @Override
+  public void setContext(String context) {
+    this.context = context;
```

Comment:
My thinking is there is no way to distinguish between a user accidentally passing null when they did not mean to and a user trying to clear the context.  However if we do not accept null (users must call clearContext insead), then we can always detect the case where a user accidentally passes null and fail fast.